src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -827,17 +827,18 @@ public INamedTypeSymbol GetTypeByMetadataName(string fullyQualifiedMetadataName)
         protected abstract INamedTypeSymbol CommonGetTypeByMetadataName(string metadataName);
 
         /// <summary>
-        /// Returns a new INamedTypeSymbol with the given element types and (optional) element names.
+        /// Returns a new INamedTypeSymbol with the given underlying type and (optional) element names.
         /// </summary>
         /// <remarks>
         /// Since VB doesn't support tuples yet, this call will fail in a VB compilation.
+        /// Also, the underlying type needs to be tuple-compatible.
         /// </remarks>
-        public INamedTypeSymbol CreateTupleTypeSymbol(ImmutableArray<ITypeSymbol> elementTypes, ImmutableArray<string> elementNames = default(ImmutableArray<string>))
+        public INamedTypeSymbol CreateTupleTypeSymbol(INamedTypeSymbol underlyingType, ImmutableArray<string> elementNames = default(ImmutableArray<string>))
```

Comment:
Neal, I understand that you disagree with the current design, you made it very clear and we are going to reconcile this in another forum. This change follows the current design, whether you like it or not, this is the reality that we have to face. If design changes, we will react accordingly and change the API. But this change should not be held as a hostage.